actionlib_lisp/src/new_implementation/comm-state-machine.lisp

```diff
@@ -146,30 +142,31 @@
    transition-callback. If the result was processed before the 
    last status update the transition-callback gets called even
    if the state-machine doesn't change"
-  (if (and (or (eql (name (get-current-state (stm csm))) :done)
-               (process-signal (stm csm) signal))
-           (transition-cb csm))
-      (funcall (transition-cb csm))))
+  (when (and (or (eql (name (get-current-state (stm csm))) :done)
+                 (process-signal (stm csm) signal))
+             (transition-cb csm))
+    (funcall (transition-cb csm))))
 
 (defmethod update-status ((csm comm-state-machine) status)
   "If the status is not equal to the last status the comm-state-machine
    gets updated with the new status"
-  (when (not (eql (latest-goal-status csm) status))
-    (with-recursive-lock ((status-mutex csm))
-      (setf (latest-goal-status csm) status)))
-  (if (get-next-state (stm csm) status)
-        (transition-to csm status)))
+  (with-recursive-lock ((csm-mutex csm))
+    (unless (eql status :lost)
+      (setf (lost-ctr csm) 0))
+    (when (get-next-state (stm csm) status)
+      (setf (latest-goal-status csm) status))
```

Comment:
This method does not compile. I guess the error is in line 156: One closing bracket too much. @jannikb Do I see this correctly?